lib/awspec/helper/client_wrap.rb

```diff
@@ -0,0 +1,46 @@
+module Awspec::Helper
+  class ClientWrap
+    attr_reader :client, :backoff, :iteration, :backoff_limit, :symbol
+    def initialize(real_client = nil, args = {})
+      raise ArgumentError, 'Client can not be nil' if real_client.nil?
+      @client        = real_client
+      @backoff       = args.key?(:backoff) ? args[:backoff] : 0.0
+      @orig_backoff  = @backoff
+      @iteration     = args.key?(:iteration) ? args[:iteration] : 1
+      @orig_iter     = @iteration
+      @backoff_limit = args.key?(:backoff_limit) ? args[:backoff_limit] : 30.0
+      # build the symbol we'll use to compare to any errors caught in method_missing
+      # below.
+      @symbol = real_client.class.to_s.split('::').shift(2).push('Errors', 'RequestLimitExceeded').join('::').to_sym
+    end
+
+    protected
+
+    # used to capture only the "RequestLimitExceeded" error from an aws
+    # client api call.  In the case of matching it we want to try again,
+    # backing off successively each time, until the backoff_limit is reached or
+    # exceeded, in which case, the error will be re-raised and it should fail
+    # as expected.
+    def method_missing(m, *args, &block)
+      begin
+        results = client.send(m, *args, &block)
+      rescue Exception => e
```

Comment:
`rescure => e` or `rescue RuntimeError => e` ?

I think range of `Exception` is too large.

Reply:
Right, that's typically not what you want to do, but what we're looking for is an unknown error that matches a symbol (see line 14 above).  This coupled with the the line below it (28), it will only capture if the exception class matches the one we're expecting (RequestLimitExceeded).  The problem is that each aws client has their own exceptions, even for a generic one like this.  See the tests associated with the code too.